Backend/router/v1/report.ctrl.js

```diff
@@ -1,120 +1,122 @@
+    return ;
+  }
+  const reportId = parseInt(req.query.reportId);
+  if (Number.isNaN(reportId)) {
+    return res.status(400).end();
+  }
+  models.report.findAll({
+    where: {
+      id: reportId
     }
-    models.report.findAll({
-      where: {
-        id: reportId
-      }
-    }).then((result) => {
+  }).then ((result) => {
+    if (result == 0) {
+      res.json({msg: "The reportId does not exist"});
+    } else {
       res.json(result);
-    }).catch((err) => {});
-  } else {
-    res.json({
-      msg: "Can't find this report"
-    });
-  }
+    }
+  }).catch((err) => {
+    res.json({msg: "Error: failed to load this report"});
+  });
 };
 
 const deleteReport = (req, res) => {
-  const reportId = parseInt(req.params.reportId);
-  //When ID is not a number
+  if (!req.isAuthenticated()) {
+    res.status(403).json({msg: "You are not logged in"});
+    return ;
+  }
+  const reportId = parseInt(req.query.reportId);
   if (Number.isNaN(reportId)) {
-    return res.status(400).end()
+    return res.status(400).end();
   }
   models.report.destroy({
     where: {
       id: reportId
     }
   }).then((result) => {
-    res.status(204).json(result);  //no content
-  }).catch((err) => {});
+    if (result == 0) {
+      res.json({msg: "The reportId does not exist"});
+    } else {
+      res.status(204).end();
+    }
+  }).catch((err) => {
+    res.json({msg: "Error: failed to delete this report"});
+  });
 };
 
 const editReport = (req, res) => {
+  if (!req.isAuthenticated()) {
+    res.status(403).json({msg: "You are not logged in"});
+    return ;
+  }
   const reportId = parseInt(req.query.reportId);
-  //When ID is not a number
   if (Number.isNaN(reportId)) {
-    return res.status(400).end()
+    return res.status(400).end();
   }
-
   const newWhat = req.body.data[0].what;
   const newLocation = req.body.data[0].location;
   const newTime = req.body.data[0].time;
   const newWho = req.body.data[0].who;
   const newDetails = req.body.data[0].details;
 
-  if (req.isAuthenticated()) {
-    models.report.update({
-        what: newWhat,
-        location: newLocation,
-        time: newTime,
-        who: newWho,
-        details: newDetails
-      },{
-        where: {id: reportId}
-    }).then((result) => {
-      res.status(200);
-      res.json({
-        msg: "update success"
-      });
-    }).catch((err) => {
-      res.json({
-        msg: "update fail"
-      });
-    });
-  } else {
-    res.status(401);
-    res.json({
-      msg: "you are not logged in"
-    });
-  }
+  models.report.update({
+      what: newWhat,
+      location: newLocation,
+      time: newTime,
+      who: newWho,
+      details: newDetails
+    },{
+      where: {id: reportId}
+  }).then((result) => {
+    if (result == 0) {
+      res.json({msg: "The reportId does not exist"});
+    } else {
+      res.status(200).end();
+    }
+  }).catch((err) => {
+    res.json({msg: "Error: failed to edit this repart"});
```

Comment:
Same here, should we have a status code for this error case?

Reply:
humm... I think we don't need it. It will be just sending an error code to client.